api/src/main/java/net/automatalib/ts/modal/ModalTransitionSystem.java

```diff
@@ -15,46 +15,48 @@
  */
 package net.automatalib.ts.modal;
 
+import java.util.Collection;
 import java.util.Iterator;
-import java.util.Map;
 
-import net.automatalib.automata.UniversalAutomaton;
+import net.automatalib.automata.UniversalFiniteAlphabetAutomaton;
 import net.automatalib.automata.concepts.InputAlphabetHolder;
+import net.automatalib.automata.graphs.TransitionEdge;
+import net.automatalib.automata.graphs.TransitionEdge.Property;
+import net.automatalib.automata.graphs.UniversalAutomatonGraphView;
+import net.automatalib.automata.visualization.MTSVisualizationHelper;
 import net.automatalib.graphs.FiniteLTS;
-import net.automatalib.visualization.DelegateVisualizationHelper;
+import net.automatalib.graphs.UniversalGraph;
+import net.automatalib.ts.modal.transitions.ModalEdgeProperty;
 import net.automatalib.visualization.VisualizationHelper;
 
 public interface ModalTransitionSystem<S, I, T, TP extends ModalEdgeProperty>
-        extends UniversalAutomaton<S, I, T, Void, TP>, FiniteLTS<S, T, I>, InputAlphabetHolder<I> {
+        extends UniversalFiniteAlphabetAutomaton<S, I, T, Void, TP>, FiniteLTS<S, T, I>, InputAlphabetHolder<I> {
```

Comment:
Do MTSs by definition have a finite set of states and inputs, or would it make sense (on an interface level) to define MTSs with unbounded states or inputs?

Reply:
Larsen (1990) defined MTSs with possible unbounded sets of states and inputs, but the transition relation should be image-finite.

Reply:
For most algorithms it makes sense to only consider finite MTSs. How about we add an `IndefiniteModalTransitionSystem` markup interface if at any point we (or others) want to model infinite systems. The `ModalTransitionSystem` interface could then be a finite refinement (similar to the `Graph` and `IndefiniteGraph` interfaces).

Reply:
I think that is a nice abstraction without any limitations on client side. But maybe we should enforce image-finiteness, because important theorems do not hold for unbounded transition sets (and these theorems may be implicitly assumed in client code).

Reply:
By making the default `ModalTransitionSystem` finite in states and input symbols and your code requiring this finite type in the parameters, everything should be fine with the properties of the theorems and most of your code can stay the way it currently is. My only issue was the lack of an indefinite version (if people want to model such an MTS).

Reply:
Yes, I got that. I just wanted to add that maybe an infinte-state ts without that restriction on the transitions may have no use case, as many formal proofs and concepts require it.

But since it does not effect our client code, it is completely up to you.